src/Network_Equivalences-Communication.thy

```diff
@@ -62,6 +88,16 @@ lemma adapted_after_duplication:
   shows "\<currency>\<^sup>+ A \<guillemotleft> \<E> = \<currency>\<^sup>+ (A \<guillemotleft> \<E>)"
   by (simp del: distributor_def add: adapted_after_distributor)
 
+lemma transition_from_duplication:
+  assumes "\<currency>\<^sup>+ A \<rightarrow>\<^sub>s\<lparr>\<alpha>\<rparr> Q"
+  obtains n and X
+    where
+      "\<alpha> = A \<triangleright> \<star>\<^bsup>n\<^esup> X"
+    and
+      "Q = (A \<guillemotleft> suffix n \<triangleleft> X \<parallel> A \<guillemotleft> suffix n \<triangleleft> X \<parallel> \<zero>) \<parallel> \<currency>\<^sup>+ A \<guillemotleft> suffix n"
+  using assms unfolding duplication_def
+  by (fastforce elim: transition_from_distributor)
```

Comment:
It’s a bit dangerous to use a simplifying prover with no restriction to the set of simplification rules here, as it could apply `distributor_def` early, thus destroying the opportunity to apply `transition_from_distributor`. This is precisely the reason why `try0` (which you surely tried :wink:) finds only `fastforce` invocations: `auto` doesn’t work with the full simpset, but does work with `simp only: general_parallel.simps`. I suggest using this invocation of `auto`, that is, using the following proof:
```isabelle
by (auto elim: transition_from_distributor simp only: general_parallel.simps)
```

Reply:
Looks all good.